src/helpers.rs

```diff
@@ -472,6 +472,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     }
 }
 
+/// Check that the number of args is what we expect.
+pub fn check_arg_count<'tcx>(args: &[OpTy<'tcx, Tag>], n: usize) -> InterpResult<'tcx> {
+    if args.len() != n {
+        throw_ub_format!("incorrect number of arguments, got {}, needed {}", args.len(), n);
```

Comment:
It would be good if this could print the name of the function that was called incorrectly.

Reply:
When I cause this to error I seem to be able to figure out what `link_name` was, but I can still pass it in explicitly if you want.

Reply:
Which is to say, should `link_name` be passed in here or just generally be added to any error in `emulate_foreign_item`?

Reply:
Hm... can you show an example for how the error looks? You are right we don't print the function name for other errors like bad argument sizes, so it's probably not worth the effort.

Reply:
When I change it in the tests I get something like
```
error: Undefined Behavior: incorrect number of arguments, got 1, needed 0
  --> $DIR/intrinsics.rs:27:14
   |
27 |     unsafe { intrinsics::forget(); }
   |              ^^^^^^^^^^^^^^^^^^^^^^^^ incorrect number of arguments, got 0, needed 1
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
```
along with a bunch of other output.

Reply:
Okay, so it should be pretty clear which function it is talking about. Thanks!